toolz/itertoolz.py

```diff
@@ -496,6 +496,34 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def duplicate(seq, n=1):
+    """ Gets each element multiple times.
+
+    Like ``itertools.repeat`` this will repeat each element n-times.
+    However, it will do this for each element of the sequence.
+
+     Args:
+
+         seq(iterable):           The sequence to grab items from.
+
+         n(integral):             Number of repeats for each element.
+
+     Returns:
+
+         generator:               A generator of repeated elements.
+
+     Examples:
+
+         >>> list(duplicate([1, 2, 3], 2))
+         [1, 1, 2, 2, 3, 3]
+    """
+
+    assert (n >= 0), "n must be positive, but got n = " + repr(n)
```

Comment:
Do we need to have our own checks here, does `itertools.repeat` catch this for us?

If it doesn't, I think we should replace these with `ValueError`s. In general, assertion errors should be used internally when invariants are violated, but `ValueError` is more standard for function input validation. For example:

```
In [5]: math.log(-1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-10344c395964> in <module>()
----> 1 math.log(-1)

ValueError: math domain error
```

Also, when creating this message, you could do:

```python
raise ValueError('n must be positive, but got %r' % n)
```
The `%r` format unit calls `repr` on the object and then formats the result into the string. I personally find this makes it easier to read the message, especially if you want to format in many objects.